### ISSUE TITLE
After Ledger device name change, its very difficult to pair with MMM

### ISSUE THREAD

**Issue by plasmacorral** [MetaMask/metamask-mobile#4921]
### **Description**

After changing the Ledger Device name via Ledger Live, it is tough to pair with MMM. I forgot the device from MMM and Bluetooth settings by its old name, powered off and back on the Nano X.  On the first attempt, the device still shows and can actually be paired with the obsolete name.  Once the name is correctly shown, the user spends a lot of time waiting on a spinning wheel.

Ultimately, I ended rebooting the mobile device and with the Ledger device powered on and unlocked, Bluetooth connected to it from Settings>Bluetooth (rather than from MMM), and attempted to pair again and succeeded.

iPhone 13 mini
Running iOS 15.6.1
MMM build 952
Ledger Nano X firmware 2.0.2

### **Reproduction steps**

I am not certain that all these steps are needed, but this is the play by play of the video supplied:
1.  Have MMM with Ledger paired (name N666)
2. Send a tx on testnet to show it works
3. Open Ledger Live and change device name (Nano666)
4. Confirm name change on Ledger
5. Close Ledger Live
6. Open MMM and login
7. Tap account icon and select Connect hardware wallet
8. forget Ledger device
9. Select Ledger 
10. Note that device is found with old name (N666)
11. Go to settings>Bluetooth
12. Forget device by old name (N666)
13. Close "Connect Ledger" by tapping X in the top right
14. Close MMM from the app try by swiping up
15. Power off ledger from device menu
16. Restart Ledger and enter PIN
17. Open settings to display device name on Nano X screen
18. Open MMM and authenticate
19. Tap account icon
20. select "Connect Hardware Wallet"
21.  Select Ledger 
22. Notice that device list still shows prior name (N666)
23. tap continue with old device name selected
24. Tap pair on the mobile device after confirming codes match
25. accept pairing on Ledger
26. Open the Ethereum app
27.  Now paired with device under old name
28. select "Connect Hardware Wallet"
29. Select Ledger 
30. Check Settings>Bluetooth and confirm paired with correct name
31. return to MMM
32. Forget Ledger device from Connect Ledger
33. Select Ledger 
34. Note that device is now found with the correct name (Nano666)
35. tap continue
36. open Ethereum app when prompted
37. wait on spinning wheel for awhile
38. Go back to wallet view
39. Note that Ledger is not in account list
40. tap account icon
41. select "Connect Hardware wallet"
42. select Ledger
43. Note that still looking for device and no longer showing device name drop-down
44. Try to access Ledger settings and device freezes and shuts off
45. Turn on Ledger and enter PIN
46. Once unlocked, device name drop down appears again
47. tap continue again
48. watch the wheel spin in the CTA until that is no longer fun

What was not captured in the video that ultimately allowed the address to once again appear and the device to be paired under the correct name:
1. Power cycle the mobile device
2. Unlock Ledger
3. Access settings>bluetooth and tap to pair the device by its updated name
4. launch MMM and try to connect hardware again

### **Acceptance Criteria**

-   After the device name is changed, the device name drop-down in Connect Ledger should reflect the updated name after the device is forgotten and the user attempts to pair again.

Scenario: Device name change forget device and bluetooth pair with correct name
- GIVEN a user is currently paired with MMM
- WHEN a user changes the device name with Ledger Live
- AND a user forgets the device under its old name
- THEN the user should be able to quickly and easily pair the device again under the correct name

### **References**

-   https://recordit.co/eBTh1EvuWI

**Comment by plasmacorral** [comment]
I think this is another issue related to changing the device name, so I will add it as a comment to this existing ticket.

https://recordit.co/8JQYuWYWDq


With a freshly reset Ledger Nano S, I restored an SRP to the Ledger Device.
I Bluetooth paired with Ledger Live and changed the device name and installed the Ethereum app.
I then attempted to pair with MMM and notice that the previous device name was shown in the drop down.
I attempted to pair with the old name anyway.
I checked the device name on the Ledger via settings.
Once prompted to open the Ethereum App, the new device name is shown in the drop-down.
Upon successful pairing, the Next button is disabled in the AMOUNT entry of the Send flow.

Same device:
iPhone 13 mini
Running iOS 15.6.1
MMM build 952
Ledger Nano X firmware 2.0.2

**Comment by plasmacorral** [comment]
This seems to be reproducible on Android 11 with Samsung A515f.  

In this scenario, prior to the start of the video clip the Nano was reset and set up as new, with that reset the prior custom device name was reverted.  After the Ledger device name is changed, MMM does not promptly find the device to pair.  When it can eventually be seen after power cycling the Nano, it is still presented with the prior device name in the add hardware flow.

In this video, you can see the Ledger device name as `Nano X 7E06` but is shown as the prior name `Nano666` within MetaMask mobile. 
https://recordit.co/9VINgqUoKr
